eos-link-host-mali-driver

```diff
@@ -22,14 +22,75 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 EXTENSION_REF=org.freedesktop.Platform.GL.host/arm/1.6
-MALI_DIR=/usr/lib/mali400/arm-linux-gnueabihf
 
-DEST_DIR="/var/lib/flatpak/extension/$EXTENSION_REF"
-mkdir -p $(dirname "$DEST_DIR")
-ln -sf "$MALI_DIR" "$DEST_DIR"
+#
+# create_flatpak_gl_host_extension FLATPAK_DIR MALI_DRIVER
+#
+# Creates a flatpak GL host extension in $FLATPAK_DIR from $MALI_DRIVER
+#
+function create_flatpak_gl_host_extension {
+    FLATPAK_DIR=$1
+    MALI_DRIVER=$2
+
+    if test ! -d $FLATPAK_DIR; then
+        return;
+    fi
+
+    DEST_DIR=$FLATPAK_DIR/extension/$EXTENSION_REF
+
+    # Ensure destination directory
+    mkdir -p $DEST_DIR
+
+    # Create lib link for flatpak GL extension
+    ln -sr $DEST_DIR $DEST_DIR/lib
 
-if test -d /var/endless-extra/flatpak; then
-    DEST_DIR="/var/endless-extra/flatpak/extension/$EXTENSION_REF"
-    mkdir -p $(dirname "$DEST_DIR")
-    ln -sf "$MALI_DIR" "$DEST_DIR"
+    # Hardlink mali driver
+    ln -t $DEST_DIR $MALI_DRIVER
```

Comment:
Why are you hardlinking instead of symlinking? I've previously added `-f` to the symlink to make sure the link gets overwritten and points to the new target. I don't know how that works with hardlinks.

Reply:
Ok, now I see what you are doing in this part. You are no linking the extension dir to the host dir directly, but creating a separate dir for the extension, then hardlinking `lib/libMali.so` in that this directly to the host's `libMali.so`. Sorry I missed that in the beginning, that threw off most of my other comments.

On that point tho, I believe if we ever update `libMali.so` on the host, the hardlink is broken on its own inode and the extension will always point to the old data. While this may even be desirable from the perspective of guaranteeing a certain runtime version does not change, doesn't this goes against the GL.host extension semantics? FWIW, without this change the GL.host extension will always provide the same GL version from the host.